src/figuresClasses.ts

```diff
@@ -1,11 +1,113 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: () => number;
+}
+
+function roundDownHundreds(value: number): number {
+  if (value === Math.round(value)) {
+    return value;
+  }
+
+  return Math.floor(value * 100) / 100;
+}
+
+export class Triangle implements Figure {
+  public readonly shape = 'triangle' as Shape;
+
+  constructor(
+    public readonly color: Color,
+    public readonly a: number,
+    public readonly b: number,
+    public readonly c: number,
```

Comment:
The constructor parameters are being assigned to class properties manually, which is unnecessary because TypeScript does this automatically when the parameters are declared with an access modifier. You can remove lines 45-48 since the properties 'color', 'a', 'b', 'c' are already declared with 'public readonly' and will be automatically assigned.